src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -234,11 +270,25 @@ public DeploymentOptions()
             foreach (var deployOptionsProp in deploymentOptionsProperties)
             {
                 var prop = options.GetType().GetProperty(deployOptionsProp.Name);
+                var attribute = prop.GetCustomAttributes(typeof(DescriptionAttribute), true)[0];
+                var description = (DescriptionAttribute)attribute;
 
                 // Note that we set excluded object types here since dacfx has this value as null;
                 if (prop != null && deployOptionsProp.Name != "ExcludeObjectTypes")
                 {
-                    deployOptionsProp.SetValue(this, prop.GetValue(options));
+                    if (deployOptionsProp.Name == "DoNotDropObjectTypes")
+                    {
+                        deployOptionsProp.SetValue(this, prop.GetValue(options));
+                    }
+                    else
+                    {
+                        DeploymentOptionProps setProp = new DeploymentOptionProps()
+                        {
+                            value = prop.GetValue(options),
+                            description = description.Description
```

Comment:
Why is this one special and have its value be a nested DeploymentOptionProps value? If that's expected shouldn't prop.GetValue be returning the correct type of object to begin with?

And the description seems like something you should have all the properties be adding as long as you're doing the proper error checking as I mentioned above.

Reply:
prop.GetValue(options) gives the value of the property coming from the DacFx which is a single bool/int/string value. where in STS I have defined a new class to include the description attribute text value along with the above value to pass it to ADS, which made me to create a new instance with value/description to set the values to these props.

Is there a way we can directly access the description attribute text defined in the c#(in DacFx) in typeScript(ADS)? Or am i missing anything! Thanks.

Reply:
So that part is fine, I was just confused why on 281 you were setting the value directly while here on 290 you were creating this new DeploymentOptionProps object and assigning that instead. But looking at it I see that's because those types are just simple arrays instead of needing to be this props object.

Is there any reason that those aren't "DeploymentOptionProps" as well? It seems like they could be and just have the array of enum values be the value. That would simplify this logic a bit which would be nice.

Reply:
Sure, Since these two props are not coming directly from DacFx and have no description, I just didn't updated them, But keeping them in as DeploymentOptionProperty would be good, have made changes and working on test cases! will add the Iteration. Thanks